src/tree/AttachedAccountsTreeItem.ts

```diff
@@ -100,21 +102,53 @@ export class AttachedAccountsTreeItem implements IAzureParentTreeItem {
 
             if (connectionString) {
                 let treeItem: IAzureTreeItem = await this.createTreeItem(connectionString, defaultExperience);
-                if (this._attachedAccounts.find(s => s.id === treeItem.id)) {
-                    vscode.window.showWarningMessage(`Database Account '${treeItem.id}' is already attached.`)
-                } else {
-                    this._attachedAccounts.push(treeItem);
-                    if (this._keytar) {
-                        await this._keytar.setPassword(this._serviceName, treeItem.id, connectionString);
-                        await this.persistIds();
-                    }
-                }
+                await this.attachAccount(treeItem, connectionString);
             }
         } else {
             throw new UserCancelledError();
         }
     }
 
+    public async attachEmulator(): Promise<void> {
+        let connectionString: string;
+        const defaultExperience = <Experience>await vscode.window.showQuickPick(['MongoDB', 'DocumentDB', 'Graph'], { placeHolder: "Select a Database Account API...", ignoreFocusOut: true });
+        if (defaultExperience) {
+            let port: number;
+            if (defaultExperience === Experience.MongoDB) {
+                port = vscode.workspace.getConfiguration().get<number>("cosmoDB.emulator.mongoPort") || 10255;
```

Comment:
I thought `WorkspaceConfiguration.get` already returned the default it if the value didn't exist. No need to have the port hard-coded in two places if we can avoid it

Reply:
That was my understanding too, but my testing seemed to throw an error here. Specifically, it looked at the default for DocDB but not Mongo. 
A restart of VSCode seemed to use the defaults correctly. :)